posts/indu-plp-part1/plp-part1.qmd

```diff
@@ -0,0 +1,217 @@
+
+## Cohort Construction
+
+Cohorts are groups of patients defined by specific criteria that are relevant to the research question. For this task, two main cohorts need to be defined:
+
+- **Target Cohort**: This refers to the group of patients we want to make predictions for. In our case, it includes patients who have been diagnosed with hypertension. These patients serve as the starting point for our prediction timeline.
+
+- **Outcome Cohort**: This refers to the clinical event we aim to predict. In our case, it includes patients from the target cohort who are subsequently diagnosed with diabetes within a specified time window. This event marks the outcome that our model will learn to forecast.
+
+These cohort definitions are central to structuring the data pipeline, as they form the foundation for downstream tasks like feature extraction, model training, and evaluation.
+
+# Defining Cohorts using OHDSICohortExpressions.jl
+
+In the context of this research, I received a 20GB synthetic dataset that contains 1,115,000 fake patients (1,000,000 alive and 115,000 deceased), each with 3 years of medical history. This dataset was provided as a DuckDB database, a lightweight, high-performance analytical database that allows fast querying of large datasets directly from local files without the need for a server.
+
+For cohort creation, I used OHDSI cohort definitions provided directly by my mentor in the form of two JSON files: **Hypertension.json** (for the target cohort) and **Diabetes.json** (for the outcome cohort). To execute them, I used the OHDSICohortExpressions.jl package to convert the JSON definitions into SQL queries, which were then run against the DuckDB database to extract the relevant cohorts.
+
+Here’s the breakdown of the process:
+
+1. Reading the cohort definitions from JSON files.
+
+2. Connecting to the DuckDB database, which stores the synthetic patient data.
+
+3. Translating the cohort definitions into SQL using OHDSICohortExpressions.jl.
+
+4. Executing the SQL queries to create the target and outcome cohorts in the database.
+
+### Cohort Definition Code
+
+Here’s how we set up the database connection and load the Parquet files into DuckDB for querying:
+
+**File:** `cohort_definition.jl`
+
+```julia
+import DBInterface: connect, execute
+import FunSQL: reflect, render
+import OHDSICohortExpressions: translate
+using DuckDB, DataFrames
+
+# Read the cohort definitions from JSON files (Hypertension and Diabetes definitions)
+target_json = read(datadir("exp_raw", "definitions", "Hypertension.json"), String)  # Target cohort (Hypertension)
+outcome_json = read(datadir("exp_raw", "definitions", "Diabetes.json"), String)  # Outcome cohort (Diabetes)
+
+# Establish a connection to the DuckDB database
+connection = connect(DuckDB.DB, datadir("exp_raw", "synthea_1M_3YR.duckdb"))
+
+# Function to process a cohort definition (translate the JSON to SQL and execute)
+function process_cohort(def_json, def_id, conn)
```

Comment:
You might need to check if the cohort table exists so add that `CREATE IF NOT EXISTS` statement here into this function just in case.